### ISSUE TITLE
Add support for proxy configuration (Socks5)

### ISSUE THREAD

**Issue by shachar-radoshinsky-am** [pika/pika#1359]
Hi,
I couldn't find a way to pass proxy information (I'm using Socks5) into the rabbit connection.
Is there a way ?

**Comment by lukebakken** [comment]
No, there is no support in Pika for SOCKS. It appears that this library may be able to accomplish it via monkey-patching -

https://pypi.org/project/PySocks/

**Comment by fetzerms** [comment]
Hi @lukebakken - I was just digging into this, as I'd like to use Pika behind a (corporate) proxy. Either Socks5 or Http/s-Proxy would be suitable for me.

Monkey-patching with PySocks (as they propose in their readme) does not seem to work, so I guess the right way would be to write an own adapter. But on the other hand, the most calls to socket seem to be in AMQPConnector, which sounds a bit "deep" for me to do some subclassing.

What way would you suggest in order to make pika connect via socks5 proxy?

**Comment by lukebakken** [comment]
> Monkey-patching with PySocks (as they propose in their readme) does not seem to work

Did you try using PySocks? I never have. If you have code to share that would be great.
